sopel/tools/__init__.py

```diff
@@ -14,26 +14,25 @@
 from __future__ import annotations
 
 import codecs
-from collections import defaultdict
 import functools
 import inspect
 import logging
 import os
 import re
 import sys
-import threading
 import traceback
-from typing import Callable
 
 from pkg_resources import parse_version
 
 from sopel import __version__
 
 from ._events import events  # NOQA
-from .identifiers import Identifier
-
-
-IdentifierFactory = Callable[[str], Identifier]
+from .identifiers import Identifier  # NOQA
+from .memories import (  # NOQA
+    SopelIdentifierMemory,
+    SopelMemory,
+    SopelMemoryWithDefault,
+)
```

Comment:
I don't really mind if we allow to do `from sopel.tools import SopelMemory, Identifier, deprecated`. The problem is when these are defined within `sopel/tools/__init__.py` because of circular import. Once they are all defined in their own submodule, there is no circular import anymore, and we are free to provide convenient shortcuts.

Reply:
> I don't really mind if we allow

Forever (is a very long time)?

Reply:
Yeah, forever. 😁

Reply:
Let's modify the suggested comment, then.

```suggestion
# shortcuts & backward compatibility with pre-8.0
from .identifiers import Identifier  # NOQA
from .memories import (  # NOQA
    SopelIdentifierMemory,
    SopelMemory,
    SopelMemoryWithDefault,
)
```